### PR TITLE
Make clippy happy and add it to CI

### LINKED ISSUE #558: clippy in CI?
All of @martinfrances107's recent PR's have got me wondering if we should incorporate clippy into our CI.

What do you think?

And if we did incorporate, should it *fail* CI or just be a warning? 

IIRC @urschrei pointed out elsewhere that sometimes these kinds of hurdles can be off-putting to new contributors who aren't already familiar with the large variety of tools.

### PR DESCRIPTION
**Pull request by nyurik**
* Add clippy to CI pipeline - might as well let CI do its magic :)
* Applied a number of fixes suggested by clippy -- tests seem to still pass, and they do make sense:
  * `&Vec<Coordinate<f64>>` -> `&[Coordinate<f64>]`
  * switch to implicit returns
  * `&String` -> `&str`
  * many boolean `assert_eq!(true, ...)` -> `assert!(...)` -- code much more compact and easy to read
  * `points.iter().copied().collect()` -> `points.to_vec()`
  * a few useless clones on copyable objects
* Allow excessive float precision in tests (complex topic, didn't want it to trigger all the time)
* Bypass a bug in rustc clippy suggestion - that code might be fixed separately if desired
* Made a tiny improvement to `geo/examples/concavehull-usage.rs` by switching `String` to `&str`

- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- ~I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.~
---

fixes #558

### CODE REVIEW

File: geo/examples/concavehull-usage.rs [hunk 1]
```diff
@@ -16,21 +16,21 @@ fn generate_polygon_str(coords: &Vec<Coordinate<f64>>) -> String {
...
         );
     }
-    return circles_str;
+    circles_str
 }
 
-fn produce_file_content(start_str: &String, mid_str: String) -> String {
-    let mut overall_string = start_str.clone();
-    overall_string.push_str(mid_str.as_ref());
+fn produce_file_content(start_str: &str, mid_str: &str) -> String {
```
Comment (lnicola) [inline geo/examples/concavehull-usage.rs:29]: This could be a `format!`, but it doesn't matter.

**Review by lnicola — APPROVED** [review]

**Comment by nyurik** [comment]
@lnicola i'm new to this proj, do i do the bors command myself, or do you (or someone else) usually does it?  Approval seems to not have triggered the auto-merge process

**Comment by lnicola** [comment]
I can approve it, but I'd prefer if someone else could take a second look, just in case I missed something.

**Comment by nyurik** [comment]
@lnicola ah, thx, just wasn't sure if i got the process right (was just reading the contribs doc) :)

File: geo/src/algorithm/closest_point.rs [hunk 2]
```diff
@@ -184,6 +184,7 @@ impl<F: GeoFloat> ClosestPoint<F> for Geometry<F> {
 
 #[cfg(test)]
 mod tests {
+    #![allow(clippy::excessive_precision)]
```
Comment (michaelkirk) [inline geo/src/algorithm/closest_point.rs:187]: Please leave a comment wherever you turn a warning off.

**Review by michaelkirk — COMMENTED** [review]
Some previous discussion here:
https://github.com/georust/geo/issues/558

The conclusion at the time was that clippy shouldn't fail CI. 

I don't have particularly strong feelings either way.

**Comment by lnicola** [comment]
I tend to agree, in another (larger) project, clippy is pretty much useless because of the false positives, so someone runs it manually from time to time.

**Review by frewsxcv — APPROVED** [review]
I feel good trying this out. If it gets in the way and Clippy complains too much about inconsequential things we can always turn it off

File: geo/src/algorithm/closest_point.rs (hunk 2 above)
Comment (nyurik) [reply geo/src/algorithm/closest_point.rs:187]: @michaelkirk what kind of a comment and where would you like me to add? thx :)

Comment (michaelkirk) [reply geo/src/algorithm/closest_point.rs:187]: Can you clarify why it is OK to turn this warning off?

Comment (nyurik) [reply geo/src/algorithm/closest_point.rs:187]: ah, got it, thx.  Well... actually I do not know this for a fact TBH. This warning is about floating point precision, which is a very tricky subject (might even be CPU dependent?). My thinking is that the floats in the tests were copy-pasted from the failed assertions, so whatever got "printed" during the failed test was expected to be the "right" result. Technically I think we should switch to using [assert_approx_eq!()](https://crates.io/crates/assert_approx_eq), but that would be a separate PR.  I will leave a comment about that.

Comment (frewsxcv) [reply geo/src/algorithm/closest_point.rs:187]: If you remove the unnecessary precision, do the tests still pass?

Comment (nyurik) [reply geo/src/algorithm/closest_point.rs]: @frewsxcv @michaelkirk ok, I reduced precision in the floats and removed the `allow` attributes, the tests seemed to have passed ok.  Feel free to merge if OK

**Pull request merged**